firstvoices/backend/views/import_job_views.py

```diff
@@ -140,117 +136,82 @@ def get_queryset(self):
-                "Please wait for it to finish before starting a new one."
-            )
+        # Queue the job for validation
+        curr_job.validation_status = JobStatus.ACCEPTED
+        curr_job.save()
 
-        # Start the task
         transaction.on_commit(
-            lambda: batch_import.apply_async(
-                (
-                    str(import_job.id),
-                    False,
-                ),
+            lambda: batch_import_dry_run.apply_async(
+                (str(import_job_id),),
                 link_error=link_error_handler.s(),
                 ignore_result=True,
             )
         )
 
-        # Update the in-memory instance and return the job
-        import_job = ImportJob.objects.get(id=import_job_id)
-        serializer = ImportJobSerializer(
-            import_job, context={"request": self.request, "site": site}
-        )
-        headers = self.get_success_headers(serializer.data)
-
-        return Response(
-            serializer.data, status=status.HTTP_202_ACCEPTED, headers=headers
-        )
+        return Response(status=status.HTTP_202_ACCEPTED)
 
     @action(detail=True, methods=["post"])
-    def validate(self, request, site_slug=None, pk=None):
-        """
-        Method to start the validation process on a given import-job.
-        """
-        site = self.get_validated_site()
+    def confirm(self, request, site_slug=None, pk=None):
         import_job_id = self.kwargs["pk"]
 
-        # Verify that no other jobs are started or queued for the same site
-        existing_incomplete_jobs = get_import_jobs_queued_or_running(
-            site, import_job_id
-        )
+        curr_job = ImportJob.objects.get(id=import_job_id)
 
-        if len(existing_incomplete_jobs):
+        if curr_job.validation_status != JobStatus.COMPLETE:
             raise ValidationError(
-                "There is at least 1 job on this site that is already running or queued to run soon. "
-                "Please wait for it to finish before starting a new one."
+                "A successful dry-run is required before doing the import. "
+                "Please validate the job before confirming the import."
             )
 
-        # Another check to prevent running validation on the current job
-        # if its already running or queued for dry run
-        curr_job = ImportJob.objects.filter(id=import_job_id)[0]
-        if curr_job.validation_status in [JobStatus.ACCEPTED, JobStatus.STARTED]:
+        if curr_job.status in [JobStatus.ACCEPTED, JobStatus.STARTED]:
             raise ValidationError(
-                "The specified job is already running or queued. "
-                "Please wait for it to finish before starting a new one."
+                "The specified job is already running or queued for import. "
+                "It also cannot be run again once the import is finished."
```

Comment:
How about, "This job has already been confirmed and is currently being imported." ?